Added Architectural overview

ihsraham opened the pull request:
Added markdown of architectural overview originally created by @akushniruk on Yellow Org Notion

ihsraham committed:
Added Architectural overview

Added markdown of architectural overview originally created by @akushniruk on Yellow Org Notion

akushniruk commented:
@ihsraham 
Let's rename it to the yellow_ecosystem.md as it's not an architecture overview for the yellow network

ihsraham committed:
Renamed the markdown

Renamed the markdown to yellow_ecosystem.md since it is more of an architecture overview for yellow ecosystem as a whole and not yellow network particularly
